File: backend/src/constants/ResourceStatus.go
package constants

const (
	ResourceAvailable   = "AVAILABLE"
	ResourceBooked      = "BOOKED"
	ResourceMaintenance = "MAINTENANCE"
	ResourceOffline     = "OFFLINE"
)

var ResourceStatuses = []string{ResourceAvailable, ResourceBooked, ResourceMaintenance, ResourceOffline}

File: frontend/src/types/ResourceStatus.ts
export const ResourceStatus = ["AVAILABLE","BOOKED","MAINTENANCE","OFFLINE"] as const;
export type ResourceStatus = (typeof ResourceStatus)[number];
export const ResourceStatusText: Record<ResourceStatus, string> = {
  AVAILABLE: "可用",
  BOOKED: "占用中",
  MAINTENANCE: "维护中",
  OFFLINE: "下线"
};
